atom/browser/web_dialog_helper.cc

```diff
@@ -182,7 +183,10 @@ class FileSelectHelper : public base::RefCounted<FileSelectHelper>,
   }
 
   // content::WebContentsObserver:
-  void WebContentsDestroyed() override { render_frame_host_ = nullptr; }
+  void WebContentsDestroyed() override {
+    render_frame_host_ = nullptr;
+    Release();
```

Comment:
If I recall correctly from when I was messing with this calling `Release()` too many times causes a `CHECK` to be fired and I tested it extensively last time to ensure that it was being released down to a ref count of 0 after each open dialog.

What change here requires that we call `Release()` again now?

Reply:
ah i should have included this in the PR body, but this is just application of the review here: https://github.com/electron/electron/pull/18343#pullrequestreview-240192999

Reply:
`FileSelectHelper` is now handed off to different sequence for directory listing by `net::DirectoryLister`, we don't want to continue the operation when the webcontents is destroyed, this is just to terminate the background operation.